code/game/machinery/_machinery.dm

```diff
@@ -831,10 +831,11 @@
 		return //Just delete us, no need to call anything else.
 
 	on_deconstruction(disassembled)
-	if(!LAZYLEN(component_parts))
+	spawn_frame(disassembled)
+
+	if(LAZYLEN(component_parts))
 		dump_contents() //drop everything inside us
 		return //we don't have any parts.
-	spawn_frame(disassembled)
```

Comment:
Please Do not touch this code. This was done so machines like the drying rack(which is a machine with no component parts) & many others don't spawn metal frames when we don't need them to.

Find another way

Reply:
They should override spawn_frame() if they don't need to spawn them.

Reply:
That's way too much work which is now your responsibility if you still want to proceed with this change. It's not logical yes because it's more of a convenience feature for machines with no parts to be deconstructed cleanly.

Leave it as it is

Reply:
Fixed it for drying rack, all good now.

Reply:
If that was the only case you can resolve this conversation 👍